src/lib/circuit_value.ts

```diff
@@ -15,22 +15,41 @@ export {
   circuitValue,
 };
 
-type Constructor<T> = { new (...args: any[]): T };
+type AnyConstructor = new (...args: any) => any;
 
 function asFieldElementsToConstant<T>(typ: AsFieldElements<T>, t: T): T {
   const xs: Field[] = typ.toFields(t);
   return typ.ofFields(xs);
 }
 
-// TODO: Synthesize the constructor if possible (bkase)
-//
 abstract class CircuitValue {
+  constructor(...props: any[]) {
```

Comment:
this function was written verbatim, completely correctly, by github copilot on the first try!!!

Reply:
That is really cool! I'm going to try out copilot.